x-pack/plugins/canvas/common/lib/hex_to_rgb.js

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const hexToRgb = hex => {
+  const shorthandHexColor = /^#?([a-f\d]{1})([a-f\d]{1})([a-f\d]{1})$/i;
+  const hexColor = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i;
+
```

Comment:
Nice regexes and approach! 

What are our requirements here, eg. do we need to support 4-element hexas with alpha of CSS4?

Reply:
The `isColorDark` service provided by EUI only checks RGB, so I'm not sure that we need to, but it'd be good to test. We also accept rgb, rgba, and color names as background colors, basically anything CSS will accept, so maybe I should support them here as well.